Firmware/util.cpp

```diff
@@ -247,36 +247,15 @@ ClCheckVersion oCheckVersion=ClCheckVersion::_None;
 ClCheckGcode oCheckGcode=ClCheckGcode::_None;
 
 void fCheckModeInit() {
-    oCheckMode = (ClCheckMode)eeprom_read_byte((uint8_t *)EEPROM_CHECK_MODE);
-    if (oCheckMode == ClCheckMode::_Undef) {
-        oCheckMode = ClCheckMode::_Warn;
-        eeprom_update_byte((uint8_t *)EEPROM_CHECK_MODE, (uint8_t)oCheckMode);
-    }
-    if (farm_mode) {
-        oCheckMode = ClCheckMode::_Strict;
-        eeprom_init_default_word((uint16_t *)EEPROM_NOZZLE_DIAMETER_uM, EEPROM_NOZZLE_DIAMETER_uM_DEFAULT);
-    }
-    oNozzleDiameter = (ClNozzleDiameter)eeprom_read_byte((uint8_t *)EEPROM_NOZZLE_DIAMETER);
-    if ((oNozzleDiameter == ClNozzleDiameter::_Diameter_Undef) && !farm_mode) {
-        oNozzleDiameter = ClNozzleDiameter::_Diameter_400;
-        eeprom_update_byte((uint8_t *)EEPROM_NOZZLE_DIAMETER, (uint8_t)oNozzleDiameter);
-        eeprom_update_word((uint16_t *)EEPROM_NOZZLE_DIAMETER_uM, EEPROM_NOZZLE_DIAMETER_uM_DEFAULT);
-    }
-    oCheckModel = (ClCheckModel)eeprom_read_byte((uint8_t *)EEPROM_CHECK_MODEL);
-    if (oCheckModel == ClCheckModel::_Undef) {
-        oCheckModel = ClCheckModel::_Warn;
-        eeprom_update_byte((uint8_t *)EEPROM_CHECK_MODEL, (uint8_t)oCheckModel);
-    }
-    oCheckVersion = (ClCheckVersion)eeprom_read_byte((uint8_t *)EEPROM_CHECK_VERSION);
-    if (oCheckVersion == ClCheckVersion::_Undef) {
-        oCheckVersion = ClCheckVersion::_Warn;
-        eeprom_update_byte((uint8_t *)EEPROM_CHECK_VERSION, (uint8_t)oCheckVersion);
-    }
-    oCheckGcode = (ClCheckGcode)eeprom_read_byte((uint8_t *)EEPROM_CHECK_GCODE);
-    if (oCheckGcode == ClCheckGcode::_Undef) {
-        oCheckGcode = ClCheckGcode::_Warn;
-        eeprom_update_byte((uint8_t *)EEPROM_CHECK_GCODE, (uint8_t)oCheckGcode);
-    }
+    oCheckMode = (ClCheckMode)eeprom_init_default_byte((uint8_t *)EEPROM_CHECK_MODE, farm_mode ? (uint8_t) ClCheckMode::_Strict : (uint8_t)ClCheckMode::_Warn);
```

Comment:
I don't think this change is completely correct. We can't rely on farm_mode during eeprom initialization since farm mode will most likely not be enabled during the first eeprom initialization, so the default value would always be loaded instead.
The old code also seems to enforce a few of these settings even if the eeprom is already initialized. The new code only sets defaults when uninitialized. I think we should try to preserve the old behaviour.

The EEPROM_NOZZLE_DIAMETER_uM_DEFAULT initialization looks as if it behaves correctly, but I'm a bit confused by the old code. It's overly complicated and I still don't understand why we need two eeprom "nozzle diameter" settings.

The EEPROM_CHECK_MODEL, EEPROM_CHECK_VERSION and EEPROM_CHECK_GCODE code looks correct 👍.

Reply:
Regarding `EEPROM_CHECK_MODE`, does the change below look OK? Surprisingly there is no change in memory usage.

The code below will always at least initialise the EEPROM value. But if farm mode is enabled, it's overwritten if it hasn't been done already.
![image](https://user-images.githubusercontent.com/8218499/231295308-f6d8c4ff-9442-467d-9f4c-797f5b6ca49f.png)

Reply:
I think that change is correct, but I didn't check it too carefully.
There is still EEPROM_NOZZLE_DIAMETER below it that doesn't look correct. We can't have a ternary operator on farm_mode in one of those eeprom _init_default functions.

Reply:
I havent forgotten the nozzle setting. Just needed a fresh 🧠 and sleep. I need to re-review how these nozzle eeprom settings are used. It’s very confused at only a quick glance. Needs a more thorough “glance” ;) 

I may delay this for the weekend to priorities the menu testing. But will see how the week goes. :)

Reply:
Hopefully resolved with [bc4089f](https://github.com/prusa3d/Prusa-Firmware/pull/4027/commits/bc4089fe6517803a8fb11be7ff33b5a531533ce8)